azclishell/az_completer.py

```diff
@@ -284,15 +296,20 @@ def gen_global_param_completions(self, text):
                         and not text.split()[-1].startswith('--') and \
                         param.startswith('-') and not param.startswith('--') and\
                         self.validate_completion(param, text.split()[-1], text, double=False):
-                    yield Completion(param, -len(text.split()[-1]))
+                    yield Completion(
+                        param, -len(text.split()[-1]),
```

Comment:
`-len(text.split()[-1]`

This is really confusing. Why the negative?  Is there any reason why you can't handle the `-len(...)` part in the constructor of Completion so that this is easier to follow?